### ISSUE TITLE
o or O option not working

### ISSUE THREAD

**Issue by nitrogenxx** [gdh1995/vimium-c#1177]
1. go to https://www.perplexity.ai/
2. press Esc for normal mode
3. press o --or-- O and type the query there

it will not let you type the query 
instead when you press  o or O it will go into insert mode while Vomnibar is still open
Vomnibar is fade out ( or transparent)

same behaviour just like you open Vomnibar and click somewhere else while Vomnibar is still open.. the Vomnibar appearance will fade out ( or transparent)

- browser : Thorium 124.0.6367.218 stable latest

- Vimium C : Version 2.11.1 latest 

- OS name and version: i use arch btw.. that's why I'm still single

**Comment by gdh1995** [comment]
Well it's not a bug but a security feature of Vimium C.

The website locks keyboard focus to its "Ask anything..." input box using some naive JavaScript code, which means everytime Vomnibar want to focus the webpage grabs back focus again.

Vomnibar gets partially transparent, so that users may learn it loses focus "at once". In the below snapshot, there's a green text caret inside the red ellipse:
![image](https://github.com/user-attachments/assets/f4c850b0-f80b-4171-a6ba-9d17fc9d29cc)

I use the word `security feature` because in such a situation Vimium C can not make Vomnibar grab back focus again and again - otherwise a web page may run into a dead loop and then eat much CPU time and laptop power.

**Comment by afilp** [comment]
Is there a way to have the "o" or any other key show only the open tabs and bookmark tabs, without history?

With the top results to be the open tabs?

I am trying to use vimium-c instead of the Chrome ctrl-shift-A, but I can't because there are too many results because of the history. 

Even better:
Can we have some options to customze the "o" view? (what to show/how/etc.) And these options be persistent?

**Comment by gdh1995** [comment]
1. ```map o Vomnibar.activate mode="omni" engines="search,tab,bookmark"```
2. No Vimium C doesn't ensure this, because suggestions are sorted by their scores computed using your query words and their last-visit-time
  - but you may type `:t <some-query>` to limit search range into `tabs` temporarily
3. You may style Vomnibar dialog by writing CSS code
  - see https://github.com/gdh1995/vimium-c/wiki/Style-the-UI-of-Vimium-C-using-Custom-CSS
  - some examples:
    -  https://github.com/gdh1995/vimium-c/issues/1170#issuecomment-2238395941

**Comment by nitrogenxx** [comment]
Vomnibar is great.. also there is one feature which is so much essential to have.. is to search on multiple website with one query 

example
`my_query="my query xyz"`

https://github.com/search?q= `$my_query`
https://www.google.com/search?q= `$my_query`
https://www.youtube.com/results?search_query= `$my_query`

boooom search on multiple website in just one shot

searching same query on multiple website is so much time consuming and mind taking slow process 
Vomnibar deserve this feature because its so much handy when you need to research about any topic

**Comment by gdh1995** [comment]
Yes Vimium C supports this, but I recommend a simpler (at least for me) way:
- you may use `searchInAnother` to switch among different search engines
- Vimium C will auto find you query words by analysing tab URLs, and then build a new URL using a given keyword

For example:
```
run <v-sia> searchInAnother##keyword=$s mask
## `GG` will auto find query words and then search in the `g` keyword
run GG sia g
## redo search in the `d` keyword
run Gd sia d
```

If you do want to open multiple URLs in one action, then you may add such a search engine:
```
multi: vimium://urls/gh|g|yt:$S
```

And then `multi foo bar` + `<Enter>` on Vomnibar will search in `gh`, `g` and `yt` custom search engines.

**Comment by nitrogenxx** [comment]
thankyou. this made my life so much easier and I feel more confident using Vimium now. the way you explained it with examples cleared all my doubts

however there is one more question i have.

how about using keyboard shortcut to open a URL and if that URL is already open somewhere in the tabs. then switch to that tab instead of reopening the same URL into new tab

i can think the can be done with two ways but i don't know how. **example**:

**1st way**: using keybinding like
when we press `alt + g` `c` it should open `https://github.com/gdh1995/vimium-c/` but when this URL is already present in some other tab then `alt + g` `c` should switch to that tab and should not open this URL again into new tab

**2nd way**: using vominibar
sort the vominibar primarily for open tabs and secondary for history.
this way when we will type `vimium-c` in the vominibar and `press enter` then if `https://github.com/gdh1995/vimium-c/` is already present in tabs it will switch to that tab otherwise it will open that into new tab

this is helpful when we have many tabs open and we don't want to search it on tab list.. this way we don't have to perform **if / else** into our mind.. kind of cool..uh.💕

**Comment by gdh1995** [comment]
1. You may add
```
# focusOrLaunch is an alias of `openUrl reuse="reuse"`
map <a-g>c focusOrLaunch url="https://github.com/gdh1995/vimium-c/"
```

2. When Vomnibar is in "omni" mode, it shows suggestions from history items, bookmarks and opened tabs, so you may press ArrowDown to select a tab suggestion:
![image](https://github.com/user-attachments/assets/75db697f-1d23-41a1-93e3-0bda43ebe79b)

Or you may try the `T` mapping (Shift+T, `Vomnibar.activateTabs`) to search in only tabs.
